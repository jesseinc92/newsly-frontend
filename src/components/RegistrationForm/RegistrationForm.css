.RegistrationForm {
  margin-top: 7rem;
}

.RegistrationForm h2 {
  font-family: 'Fjalla One', sans-serif;
}

.headline-group-goal span {
  display: inline-block;
  width: 75%;
  text-align: left;
  color: #495057;
  margin-bottom: 2rem;
}

.goal-label {
  position: relative;
  border: 2px solid black;
  border-radius: 1rem;
  margin-top: 2rem;
  height: 5rem;
  padding: 0.75rem;
  text-align: left;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: box-shadow linear;
}

input[type='radio']:checked + .goal-label {
  box-shadow: inset 0 0 0 5rem black;
  color: white;
}

.RegistrationForm input[type='radio'] {
  display: none;
}

.RegistrationForm-submit {
  background-color: black;
  color: white;
  font-size: 1.25rem;
  font-family: 'taviraj', serif;
  font-weight: 400;

  border: 2px solid black;
  border-radius: 1.5rem;
  padding: 0.25rem 1.5rem;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

/* Laptop/Desktop sizing */
@media screen and (min-width: 1024px) {

  .RegistrationForm h2 {
    font-size: clamp(30px, 2.9vw, 48px);
  }

}