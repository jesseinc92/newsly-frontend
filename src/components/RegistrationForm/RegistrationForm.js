import React, { useContext, useState } from "react";
import { Link, Navigate, useNavigate } from 'react-router-dom';
import useFormChange from "../../hooks/useFormChange";
import UserContext from "../../UserContext";
import NewslyAPI from '../../api';
import './RegistrationForm.css';
import Button from "../Button/Button";

const RegistrationForm = () => {
  const { user, setUser } = useContext(UserContext);
  const [loginPage, setLoginPage] = useState(1);
  const redirect = useNavigate();

  const [formData, handleChange] = useFormChange({
    firstName: '',
    lastName: '',
    username: '',
    password: '',
    goal: ''
  });

  const handleRegister = async e => {
    e.preventDefault();
    try {
      await NewslyAPI.registerUser(formData);

      // API call to get user
      const loggedUser = await NewslyAPI.getUser(formData.username);
      localStorage.setItem('user', JSON.stringify(loggedUser));
      setUser(loggedUser);

      // Once form is submitted redirect back to homepage.
      redirect('/')
    } catch(err) {
      alert(err)
      setLoginPage(1);
    }
  }

  if (!user) {
    if (loginPage === 1) {
      return (
        <section className='RegistrationForm'>
          <h2>Register today to become more mindful.</h2>
          <p>Already have an account? <Link className='underlined-link' to='/user/login'>Login</Link> here.</p>
          <form>
            <div className='input-group'>
              <label htmlFor='firstName'>First Name</label>
              <input 
                id='firstName' 
                name='firstName' 
                type='text' 
                value={formData.firstName} 
                onChange={handleChange} 
                autoFocus 
              />
            </div>

            <div className='input-group'>
              <label htmlFor='lastName'>Last Name</label>
              <input 
                id='lastName' 
                name='lastName' 
                type='text' 
                value={formData.lastName} 
                onChange={handleChange} 
              />
            </div>

            <div className='input-group'>
              <label htmlFor='username'>Username</label>
              <input 
                id='username' 
                name='username' 
                type='text' 
                value={formData.username} 
                onChange={handleChange}  
              />
            </div>

            <div className='input-group'>
              <label htmlFor='password'>Password</label>
              <input 
                id='password' 
                name='password' 
                type='password' 
                value={formData.password} 
                onChange={handleChange}
               />
            </div>
          <Button text='Next' handler={() => setLoginPage(2)} />
          </form>
        </section>
      )
    }

    if (loginPage === 2) {
      return (
        <section className='RegistrationForm'>
          <form onSubmit={handleRegister}>
              <div className='headline-group-goal'>
                <h2>Select Your Goal</h2>
                <span>You can only select one. Don't worry, you can change this later.</span>
              </div>
              <div className='input-group goal-group'>
                <input type='radio' id='all' name='goal' value='all' onChange={handleChange} />
                <label className='goal-label' htmlFor='all'>I want to keep up to speed with all types of news.</label>
              </div>
              <div className='input-group goal-group'>
                <input type='radio' id='usNews' name='goal' value='usNews' onChange={handleChange} />
                <label className='goal-label' htmlFor='usNews'>US News is where I want to spend my time.</label>
              </div>
              <div className='input-group goal-group'>
                <input type='radio' id='worldNews' name='goal' value='worldNews' onChange={handleChange} />
                <label className='goal-label' htmlFor='worldNews'>I need to read more World News.</label>
              </div>
              <div className='input-group goal-group'>
                <input type='radio' id='business' name='goal' value='business' onChange={handleChange} />
                <label className='goal-label' htmlFor='business'>I'd like to read more about business.</label>
              </div>
              <div className='input-group goal-group'>
                <input type='radio' id='opinion' name='goal' value='opinion' onChange={handleChange} />
                <label className='goal-label' htmlFor='opinion'>I'd like to know what others think about current events.</label>
              </div>
              <div className='input-group goal-group'>
                <input type='radio' id='science' name='goal' value='science' onChange={handleChange} />
                <label className='goal-label' htmlFor='science'>I want to keep up with science.</label>
              </div>
              <div className='input-group goal-group'>
                <input type='radio' id='culture' name='goal' value='culture' onChange={handleChange} />
                <label className='goal-label' htmlFor='culture'>Cultural news matters more to me.</label>
              </div>
              <div className='input-group goal-group'>
                <input type='radio' id='sports' name='goal' value='sports' onChange={handleChange} />
                <label className='goal-label' htmlFor='sports'>I want to be up to date on the latest sports news.</label>
              </div>
              <div className='input-group goal-group'>
                <input type='radio' id='lifestyle' name='goal' value='lifestyle' onChange={handleChange} />
                <label className='goal-label' htmlFor='lifestyle'>Lifestyle is where I want to spend my time.</label>
              </div>
            <input className='RegistrationForm-submit' type='submit' value='Register' />
          </form>
        </section>
      )
    }
  } else {
    return <Navigate to='/' />
  }
}

export default RegistrationForm;