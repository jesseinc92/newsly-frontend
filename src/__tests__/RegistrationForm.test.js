import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderWithContext from '../helpers/testContext';
import RegistrationForm from '../components/RegistrationForm/RegistrationForm';

it('renders the component', function() {
  renderWithContext(<MemoryRouter><RegistrationForm /></MemoryRouter>, null)
});

it('matches the snapshot', function() {
  const { asFragment } = renderWithContext(<MemoryRouter><RegistrationForm /></MemoryRouter>, null);
  expect(asFragment()).toMatchSnapshot();
})