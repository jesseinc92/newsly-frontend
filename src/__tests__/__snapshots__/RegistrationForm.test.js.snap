// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="RegistrationForm"
  >
    <h2>
      Register today to become more mindful.
    </h2>
    <p>
      Already have an account? 
      <a
        class="underlined-link"
        href="/user/login"
      >
        Login
      </a>
       here.
    </p>
    <form>
      <div
        class="input-group"
      >
        <label
          for="firstName"
        >
          First Name
        </label>
        <input
          id="firstName"
          name="firstName"
          type="text"
          value=""
        />
      </div>
      <div
        class="input-group"
      >
        <label
          for="lastName"
        >
          Last Name
        </label>
        <input
          id="lastName"
          name="lastName"
          type="text"
          value=""
        />
      </div>
      <div
        class="input-group"
      >
        <label
          for="username"
        >
          Username
        </label>
        <input
          id="username"
          name="username"
          type="text"
          value=""
        />
      </div>
      <div
        class="input-group"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
      <button
        class="Button"
      >
        <a
          href="/"
        >
          Next
        </a>
      </button>
    </form>
  </section>
</DocumentFragment>
`;
